### PR TITLE
docs(rxPaginate): FRMW-188-Migrate rxPaginate to API Docs

### PR DESCRIPTION
**Pull request by 100stacks**
JIRA: https://jira.rax.io/browse/FRMW-188

- [x] Dev LGTM
- [x] Dev LGTM

Migrate `rxPaginate` to API Documentation

### CODE REVIEW

**Comment by 100stacks** [comment]
Please let me know if we need to breakout the overview into `rxPaginate` services/directives/filters sections.  Thanks.

File: src/rxPaginate/rxPaginate.js [hunk 1]
```diff
@@ -4,8 +4,481 @@
...
+ * instance described above.
+ *
+ * Because all of the `servers` get passed via `ng-repeat`, it means you don't need to take explicit 
+ * action if the set of data changes. You can change `$scope.servers` at any time, and `<rx-paginate>` 
+ * will automatically re-process it.
+ *
+ * ## Persistence
+ *
+ * The user's preference for the number of items to display per page will be persisted across applications 
+ * using localstorage. This preference is set whenever the user selects a new number to show.
```
Comment (bry4n) [inline src/rxPaginate/rxPaginate.js]: localstorage vs localStorage?

File: src/rxPaginate/rxPaginate.js [hunk 2]
```diff
@@ -4,8 +4,481 @@
...
+ * `rxPaginate` module, and `pager` is a variable on your scope created like 
+ * `$scope.pager = PageTracking.createInstance();`. 
+ *
+ * This `pager` is responsible for tracking pagination state (i.e. "which page are we on", "how many 
+ * items per page", "total number of items tracked", etc.
+ *
+ * To add the pagination buttons to your table, do the following in your `<tfoot>`:
+ * <pre>
+ * <tfoot>
+ *   <tr class="paginate-area">
```
Comment (bry4n) [inline src/rxPaginate/rxPaginate.js]: example needs to use 4 space

```html
<pre>
<tfoot>
    <tr>
        <td> ... </td>
    </tr>
</tfoot>
</pre>
```

File: src/rxPaginate/rxPaginate.js [hunk 3]
```diff
@@ -4,8 +4,481 @@
...
+ * You can use this code on any part of your view. For example, if you have pagination in your table 
+ * footer, it's a good idea to hide the entire footer:
+ *
+ * <pre>
+ * <tfoot ng-hide="pager.totalPages === 1">
+ *   <tr class="paginate-area">
+ *     <td colspan="12">
+ *       <rx-paginate page-tracking="pager"></rx-paginate>
+ *      </td>
+ *   </tr>
```
Comment (bry4n) [inline src/rxPaginate/rxPaginate.js]: 4 spaces

File: src/rxPaginate/rxPaginate.js [hunk 4]
```diff
@@ -4,8 +4,481 @@
...
+ * ** END WARNING ** 
+ *
+ * The `<tfoot>` will look like this:
+ *
+ * <pre>
+ * <tfoot>
+ *   <tr class="paginate-area">
+ *     <td colspan="2">
+ *       <rx-paginate 
+ *         page-tracking="pagedServers"
```
Comment (bry4n) [inline src/rxPaginate/rxPaginate.js]: 4 spaces.

```html
<td>
    <rx-paginate
        page-tracking="pagedServer"
        ...
```

File: src/rxPaginate/rxPaginate.js [hunk 5]
```diff
@@ -4,8 +4,481 @@
...
+ *   filterText: some_filter_search_text,
+ *   selections: selected_options_from_filters,
+ *   sortColumn: the_selected_sort_column,
+ *   sortDirection: the_direction_of_the_sort_column
+ * });
+ * </pre>
+ *
+ * where:
+ *
+ * - `pageNumber`: the 0-based page of data that the user has clicked on/requested
```
Comment (bry4n) [inline src/rxPaginate/rxPaginate.js]: this file is using both dash and asterisk - should we use asterisk for consistency?

File: src/rxPaginate/rxPaginate.js [hunk 6]
```diff
@@ -4,8 +4,481 @@
...
+ *
+ * Let's say that you had defined your `getItems()` method on an object called `pageRequest`,
+ *
+ * <pre>
+ * var pageRequest = {
+ *     getItems: function (pageNumber, itemsPerPage, opts) {
+ *           var defer = $q.deferred();
+ *           ...
+ *       }
+ *   }
```
Comment (bry4n) [inline src/rxPaginate/rxPaginate.js]: add semicolon.

`};`

File: src/rxPaginate/rxPaginate.js [hunk 7]
```diff
@@ -4,8 +4,481 @@
...
+ *   getItems: function (pageNumber, itemsPerPage, opts) {
+ *       var promise = this.getItemsFromAPI(pageNumber, itemsPerPage, opts);
+ * 
+ *       rxPromiseNotifications.add(promise, {
+ *           error: 'Error loading page ' + pageNumber
+ *       }
+ *
+ *       return promise;
+ *   }
+ * }
```
Comment (bry4n) [inline src/rxPaginate/rxPaginate.js]: add semicolon

File: src/rxPaginate/rxPaginate.js [hunk 8]
```diff
@@ -4,8 +4,481 @@
...
+ *
+ * <pre>
+ * $scope.watch('extraSearch', $scope.pagedServers.refresh);
+ *
+ * var serverInterface = {
+ *   getItems: function (pageNumber, itemsPerPage, opts) {
+ *       var extraSearch = $scope.extraSearch;
+ *       return callServerApi(pageNumber, itemsPerPage, opts, extraSearch);
+ *   }
+ * }
```
Comment (bry4n) [inline src/rxPaginate/rxPaginate.js]: add semicolon

File: src/rxPaginate/rxPaginate.js [hunk 9]
```diff
@@ -4,8 +4,481 @@
...
+ *       deferred.resolve({
+ *           items: items.slice(vals.offset),
+ *           pageNumber: pageNumber,
+ *           totalNumberOfItems: items.totalNumberOfItems
+ *       });
+ *
+ *   });
+ *
+ *   return deferred.promise;
+ * }
```
Comment (bry4n) [inline src/rxPaginate/rxPaginate.js]: add semicolon

File: src/rxPaginate/rxPaginate.js (hunk 2 above)
Comment (100stacks) [inline src/rxPaginate/rxPaginate.js]: @bry4n thanks for the clarification.  I adjusted for limited `<pre>` box size...I revert to 4 spaces. :+1:

File: src/rxPaginate/rxPaginate.js (hunk 5 above)
Comment (100stacks) [inline src/rxPaginate/rxPaginate.js]: @bry4n okay, I'll switch them to asterisk :ok_hand:

**Comment by bry4n** [comment]
LGTM

File: src/rxPaginate/rxPaginate.js (hunk 2 above)
Comment (CITguy) [inline src/rxPaginate/rxPaginate.js]: FYI: We do not have documented indentation standards for HTML. For future reference, we can leave HTML indentation alone until then.

**Comment by CITguy** [comment]
LGTM

**Pull request merged**